eth_account/_utils/structured_data/hashing.py

```diff
@@ -0,0 +1,212 @@
+import json
+
+from eth_abi import (
+    encode_abi,
+    is_encodable,
+)
+from eth_utils import (
+    keccak,
+    to_tuple,
+)
+
+from .validation import (
+    validate_structured_data,
+)
+
+
+def dependencies(primaryType, types):
+    """
+    Perform DFS to get all the dependencies of the primaryType
+    """
+    deps = set()
+    struct_names_yet_to_be_expanded = [primaryType]
+
+    while len(struct_names_yet_to_be_expanded) > 0:
+        struct_name = struct_names_yet_to_be_expanded.pop()
+
+        deps.add(struct_name)
+        fields = types[struct_name]
+        for field in fields:
+            # If this struct type has already been seen, then don't expand on it'
+            if field["type"] in deps:
+                continue
+            # If this struct type is not a customized type, then no need to expand
+            elif field["type"] not in types:
+                continue
+            # Custom Struct Type
+            else:
+                struct_names_yet_to_be_expanded.append(field["type"])
+
+    # Don't need to make a struct as dependency of itself
+    deps.remove(primaryType)
+
+    return tuple(deps)
+
+
+def dict_to_type_name_converter(field):
```

Comment:
@pipermerriam this function doesn't just give the `field name`, but it also includes the `field type`.

So how about we go with `construct_field_representation`? Since what it is returning is the string representation of the `field`.

Reply:
does the term `identifier` work well here?

Reply:
@pipermerriam how about `field_identifier`? (Because I have used `IDENTIFIER_REGEX` which  checks that `field["name"]` follows a pattern.)